test/index.js

```diff
@@ -108,6 +108,43 @@ describe('DDWAF', () => {
     assert.throws(() => waf.updateRuleData([]), new Error('Could not update rule data on a disposed WAF'))
   })
 
+  it('should refuse to toggle rules with bad signature', () => {
+    const waf = new DDWAF(rules)
+    assert.throws(() => waf.toggleRules(), new Error('Wrong number of arguments, expected 1'))
+    assert.throws(() => waf.toggleRules(73), new TypeError('First argument must be an object'))
+  })
+
+  it('should not collect an attack on a toggled off rule', () => {
+    const waf = new DDWAF(rules)
+    const context = waf.createContext()
```

Comment:
Let's use two different wafContexts for `resultToggledOff` and `resultToggledOn` because sometimes rules only match 1 time per context and that will create false negatives.